client/container_list.go

```diff
@@ -35,7 +37,14 @@ func (cli *Client) ContainerList(ctx context.Context, options types.ContainerLis
 	}
 
 	if options.Filter.Len() > 0 {
-		filterJSON, err := filters.ToParam(options.Filter)
+		filterJSON := ""
+		err := errors.New("")
+		// for daemons older than v1.10, filter must be of the form map[string][]string
+		if cli.version != "" && version.Version(cli.version).LessThan(version.Version("1.10")) {
```

Comment:
@dongluochen The problem I'm facing is that one client (with a constant API version) will make a call to the Swarm master, which will then make a call via `engine-api` to a daemon that could be old or new. In such a situation, how does Swarm indicate to the client what API version to use?

@calavera Not an issue. I didn't do that right now because `ToParam` is exported, so I wasn't sure we wanted to change it. WDYT?